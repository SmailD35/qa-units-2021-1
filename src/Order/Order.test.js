jest.mock('../utils/getDate');

import React from 'react'
import {shallow, configure} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Order from './Order';
import {getDate} from "../utils/getDate";
Enzyme.configure({ adapter: new Adapter() })

afterEach(() => {
  jest.clearAllMocks();
});

describe('Order.js', () => {
  it('order without items', () => {
    getDate.mockReturnValue('14 марта, чт, 2019 год');

    const order = {
      date: 1552585550000,
      shop: 'Эльдоградо'
    };

    const output = shallow(
        <Order
            order={order}
        />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
    expect(getDate).toBeCalledTimes(1);
  });

  it('order with items', () => {
    getDate.mockReturnValue('13 марта, ср, 2019 год');

    const order = {
      date: 1552585550000,
      shop: 'Lamodник.ru',
      items: [
        'Жакет - BOREAL5',
        'Miss Gabby Костюм',
        'Ostin перчатки мужские',
        'Zara худи роз.',
      ]
    };

    const output = shallow(
        <Order
            order={order}
        />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
    expect(getDate).toBeCalledTimes(1);
  });

  it('order without date and shop', () => {
    const order = {
      items: [
        'Жакет - BOREAL5',
        'Miss Gabby Костюм',
        'Ostin перчатки мужские',
        'Zara худи роз.',
      ]
    };

    const output = shallow(
        <Order
            order={order}
        />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('order is undefined', () => {
    const output = shallow(
        <Order/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
