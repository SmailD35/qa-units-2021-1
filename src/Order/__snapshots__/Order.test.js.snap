// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js order is undefined 1`] = `""`;

exports[`Order.js order with items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Lamodник.ru
    </span>
    <span
      id="date"
    >
      13 марта, ср, 2019 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Жакет - BOREAL5
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Miss Gabby Костюм
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Ostin перчатки мужские
    </div>
    <div
      className="Order-item"
      key="3"
    >
      Zara худи роз.
    </div>
  </div>
</div>
`;

exports[`Order.js order without date and shop 1`] = `""`;

exports[`Order.js order without items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Эльдоградо
    </span>
    <span
      id="date"
    >
      14 марта, чт, 2019 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;
